pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -56,12 +56,16 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
+		if _, ok := pod.Annotations["sidecar.istio.io/status"]; !ok {
+			continue
+		}
+		if !isPodReady(pod){
+			continue
+		}
 		for _, container := range pod.Spec.Containers {
-			containsPrefix := strings.Contains(container.Image, image.Prefix)
+			containsIstioProxyV2Prefix := strings.Contains(container.Image, image.Prefix)
```

Comment:
Shouldn't this name stay generic? Keep in mind that `image` property is still called `Prefix` (accessed by `image.Prefix`) - nothing about Istio proxyv2.